### PR TITLE
Bart support scaled_dot_product_attention

### PR DESCRIPTION
**Pull request by fxmarty**
As per title.

### CODE REVIEW

File: optimum/bettertransformer/models/decoder_models.py [hunk 1]
```diff
@@ -476,10 +475,6 @@ def project(hidden_states, proj_layer, key_value_states, past_key_value):
                 attn_weights, p=self.layer.dropout, training=self.layer.training
             )  # (batch_size, n_heads, seq_length, key_length)
 
-            # Mask heads if we want to
```
Comment (younesbelkada) [inline optimum/bettertransformer/models/decoder_models.py:479]: Why these have been removed?

File: optimum/bettertransformer/models/decoder_models.py [hunk 2]
```diff
@@ -491,3 +486,94 @@ def project(hidden_states, proj_layer, key_value_states, past_key_value):
         if output_attentions:
             outputs = outputs + (attn_weights,)
         return outputs
+
+
+class BartAttentionLayerBetterTransformer(BetterTransformerBaseLayer):
+    def __init__(self, layer: "nn.Module", config: "PretrainedConfig"):
+        super().__init__(config, layer)
+
+        self.layer = layer
```
Comment (younesbelkada) [inline optimum/bettertransformer/models/decoder_models.py:495]:
```suggestion
        self.layer = layer
        self.is_decoder = True
```
no?

File: tests/bettertransformer/testing_utils.py [hunk 3]
```diff
@@ -89,32 +89,34 @@ def _test_fp16_inference(self, model_id: str, use_to_operator=False, **preproces
...
 
+        length = 50
         with torch.no_grad():
             r"""
             Make sure the models are in eval mode! Make also sure that the original model
             has not been converted to a fast model. The check is done above.
             """
             torch.manual_seed(0)
-            output_hf = hf_random_model.generate(**inputs)
+            output_hf = hf_random_model.generate(**inputs, min_length=length, max_length=length)
```
Comment (younesbelkada) [inline tests/bettertransformer/testing_utils.py:112]: I don't really see the advantage of using `auto_model_class` since here we always call `generate`

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for this! 🚀

File: optimum/bettertransformer/models/decoder_models.py (hunk 1 above)
Comment (fxmarty) [reply optimum/bettertransformer/models/decoder_models.py:479]: This is not supported and checked for above

Comment (younesbelkada) [reply optimum/bettertransformer/models/decoder_models.py:479]: Thanks makes sense!

File: tests/bettertransformer/testing_utils.py (hunk 3 above)
Comment (younesbelkada) [reply tests/bettertransformer/testing_utils.py:112]: Makes sense! Thanks for clarifying
